src/amo/components/AddonMoreInfo/index.js

```diff
@@ -73,79 +134,45 @@ export class AddonMoreInfoBase extends React.Component {
-            }
-          )}
-        </dd>
-        {addon.current_version.license ? (
+        <dd>{versionLastUpdated}</dd>
+        {versionLicenseLink ? (
           <dt className="AddonMoreInfo-license-title">
             {i18n.gettext('License')}
           </dt>
         ) : null}
-        {addon.current_version.license ? (
-          <dd>
-            <a
-              className="AddonMoreInfo-license-link"
-              href={addon.current_version.license.url}
-            >
-              {addon.current_version.license.name}
-            </a>
-          </dd>
-        ) : null}
-        {addon.has_privacy_policy ? (
+        {versionLicenseLink ? <dd>{versionLicenseLink}</dd> : null}
+        {privacyPolicyLink ? (
           <dt className="AddonMoreInfo-privacy-policy-title">
             {i18n.gettext('Privacy Policy')}
           </dt>
         ) : null}
-        {addon.has_privacy_policy ? (
-          <dd>
-            <Link
-              className="AddonMoreInfo-privacy-policy-link"
-              href={`/addon/${addon.slug}/privacy/`}
-            >
-              {i18n.gettext('Read the privacy policy for this add-on')}
-            </Link>
-          </dd>
-        ) : null}
-        {addon.has_eula ? (
+        {privacyPolicyLink ? <dd>{privacyPolicyLink}</dd> : null}
+        {eulaLink ? (
           <dt className="AddonMoreInfo-eula-title">
             {i18n.gettext('End-User License Agreement')}
           </dt>
         ) : null}
-        {addon.has_eula ? (
-          <dd>
-            <Link
-              className="AddonMoreInfo-eula-link"
-              href={`/addon/${addon.slug}/eula/`}
-            >
-              {i18n.gettext('Read the license agreement for this add-on')}
-            </Link>
-          </dd>
-        ) : null}
-        {addon.id ? (
-          <dt
-            className="AddonMoreInfo-database-id-title"
-            title={i18n.gettext(`This ID is useful for debugging and
-              identifying your add-on to site administrators.`)}
-          >
-            {i18n.gettext('Site Identifier')}
-          </dt>
-        ) : null}
-        {addon.id ? (
-          <dd className="AddonMoreInfo-database-id-content">
-            {addon.id}
-          </dd>
-        ) : null}
+        {eulaLink ? <dd>{eulaLink}</dd> : null}
+        <dt
+          className="AddonMoreInfo-database-id-title"
+          title={i18n.gettext(`This ID is useful for debugging and
+            identifying your add-on to site administrators.`)}
+        >
+          {i18n.gettext('Site Identifier')}
+        </dt>
+        <dd className="AddonMoreInfo-database-id-content">
+          {addonId}
+        </dd>
+        <dt className="AddonMoreInfo-version-history-title">
+          {i18n.gettext('Version History')}
+        </dt>
+        <dd>{versionHistoryLink}</dd>
+        <dt className="AddonMoreInfo-dev-channel-title">
+          {i18n.gettext('Development Channel')}
```

Comment:
I like the less jarring transition, but for the Beta Channel you might need to do a check to make sure `current_beta_version` exists. As I understand it not all add-ons have/require a beta version, so it'd be weird if this header appeared with the loading content then went away.